avocado/core/nrunner.py

```diff
@@ -18,6 +18,8 @@
 import unittest
 from uuid import uuid1
 
+from .test_id import TestID
```

Comment:
This breaks the current independence of the nrunner most basic runners on Avocado.  For instance, this would break:

```
avocado run --test-runner=nrunner --nrunner-spawner=podman -- /bin/true
```

I'm not against reverting the decision to make `avocado.core.nrunner` (and thus `avocado-runner-exec`, `avocado-runner-exec-test`, `avocado-runner-python-unittest`, etc) depend on Avocado... but we must come up with an alternative solution.

If we decide to always require Avocado, something similar to what is triggered by [this](https://github.com/avocado-framework/avocado/blob/ead6705d7d7a6ade80d368853f1a909760df3abe/examples/tests/passtest_explicit_requirement.py#L9) requirement tag could be assumed.

But, maybe there's another simpler solution here...

Reply:
@richtja, @clebergnu:

As I said before, I'm in favor of trying to make Avocado part of the bundle that we ship to the runners, to have a better code reuse. However, I agree with @clebergnu here. Maybe we could discuss better this on a different topic and try to find another solution here.

@richtja how difficult it is to store only the string instead of a TestID object ? I mean: how dependent we are on this object?  Looks like this object has only this property and all other properties are being used to calculate this str representation.

Reply:
Hi @beraldoleal, @clebergnu sorry I didn't realize this problem. I will look for another solution.

> @richtja how difficult it is to store only the string instead of a TestID object ? I mean: how dependent we are on this object? Looks like this object has only this property and all other properties are being used to calculate this str representation.

Unfortunately the `human`, `tap` and `testlog` plugins are dependent on TestID object. I will have to create some adapter for them.